src/mono/browser/runtime/diagnostics/diag-ws.ts

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { IDiagConnection, DiagConnectionBase, diagnostic_server_loop, schedule_diagnostic_server_loop } from "./common";
+
+export function createDiagConnectionWs (socket_handle:number, url:string):IDiagConnection {
+    return new DiagConnectionWS(socket_handle, url);
+}
+
+// this is used together with `dotnet-dsrouter` which will create IPC pipe on your local machine
+// 1. run `dotnet-dsrouter server-websocket` this will print process ID and websocket URL
+// 2. configure your wasm dotnet application `.withEnvironmentVariable("DOTNET_DiagnosticPorts", "ws://127.0.0.1:8088/diagnostics")`
```

Comment:
Is this something that needs to go in the user app? Do we need to include this in `launchSettings.json` on the templates and somehow flow it to wasm?

Reply:
`ds_router` is process external to the browser process. It could even be on different machine.
The whole websocket/ds_router experience is "nice to have" priority.

In my opinion, the ideal user experience is to improve dev-server to contain ds-router equivalent.
That's also "nice to have" priority.
I may return to it after my other goals for Net10 are done.

Reply:
Do we want to have a tracking issue for it?

Reply:
It's already in https://github.com/dotnet/runtime/issues/76316